Create connect-to-surrealdb-via-ngrok-tunnel.mdx (#290)
surrealdb/docs.surrealdb.com: versioned_docs/version-1.1.x/how-to/connect-to-surrealdb-via-ngrok-tunnel.mdx

## Connect to SurrealDB via Ngrok tunnel

This comprehensive guide will walk you through connecting a local SurrealDB instance to the internet using Ngrok, making it accessible remotely. 

> Ngrok is **a cross-platform application that allows developers to expose their local web servers to the internet**. It hosts a local web server on its own sub-domain and makes your local development box available on the internet through Tunneling.

Author: I would like to not specifically claim that on our guide. In the end it depends on the user. Connecting to SurrealDB via Ngrok should only be used for Testing purposes and not production.

> Ngrok is **a cross-platform application that allows developers to expose their local web servers to the internet**. It hosts a local web server on its own sub-domain and makes your local development box available on the internet through Tunneling.

Reviewer: Sounds good. Let's add a link to Ngrok and resolve this conversation.

> This guide has been tested for v1.1.x of SurrealDB.

### Prerequisites

This guide assumes the following:

- You have [SurrealDB installed](https://surrealdb.com/install) and a [fresh instance of SurrealDB running.](/docs.surrealdb.com/docs/introduction/start)
- Downloaded [Ngrok](https://ngrok.com/download) to be used as a tunnel.

> Tunneling refers to the process of using a network protocol to encapsulate a different payload protocol, enabling data to pass securely through a network or the internet
> 

### Steps

1. **Start SurrealDB Instance**
    
    Open your command line or terminal and run the following command to [start SurrealDB](/docs.surrealdb.com/docs/introduction/start/).
    
    ```bash
    surreal start memory -A --auth --user root --pass root
    ```
    
    We use the default username and password “root”.
    
2. **Set Up Ngrok Tunnel**
    
    Open another command line or terminal window (do not close the SurrealDB one) and run the following command to expose SurrealDB’s default port (8000) to the internet:
    
    ```bash
    ngrok http 8000
    ```
    

>     Note the forwarding address provided by Ngrok (e.g., **`25f6-2402-e280-2189-38e-9c15-d08-2f83-779e.ngrok-free.app`**).

Author: It is not a note noun, but a verb note.

    
    Keep this address handy as we will use it in the next step.
    
3. **Connect to SurrealDB via the Ngrok Address**
    
    Now, let's [connect to your SurrealDB instance](/docs.surrealdb.com/docs/cli/sql/) using the forwarding address provided by Ngrok.
    
    Run the following command in a new terminal window/tab, replacing **`[ngrok-address]`** with the actual Ngrok forwarding address you noted earlier:
    
    ```bash

>     surreal sql --conn wss://[ngrok-address] --user root --pass root --ns test --db test --pretty

Author: I have linked the guide to getting started there which explains the concepts of namespace and database in depth. The goal of this guide is not to explain each SurrealDB command unless it has never been introduced before but to reach the goal of connecting to the DB quickly. 

For anyone trying to connect to surrealDB, namespace and database won't be new concepts and if they are they should actually be redirected to its documentation to get a better understand of it.

>     surreal sql --conn wss://[ngrok-address] --user root --pass root --ns test --db test --pretty

Reviewer: I agree, but the SQL command hasn't been introduced previously. We could just add the link to the SQL command doc https://docs.surrealdb.com/docs/cli/sql in the explanation.

>     surreal sql --conn wss://[ngrok-address] --user root --pass root --ns test --db test --pretty

Author: It has already been linked on line 45.

    ```
    
4. **Verify the Connection**
    
    Finally, let’s ensure the connection is properly set up by running a test query.

>     With the help of the `CREATE` statement, [create a new record](/docs.surrealdb.com/docs/surrealql/statements/create). 

Author: I will update the link to be a relative link, but what are we trying to achieve by adding two links to the same page in a single line?
The initial link is on the action phrase ie. "create a new record". I can remove it from there and add it to `CREATE`. But it does not make sense to me to have 2 links to the same page in the same line.

>     With the help of the `CREATE` statement, [create a new record](/docs.surrealdb.com/docs/surrealql/statements/create). 

Reviewer: Oh my bad. You are correct. One link makes sense

---
sidebar_position: 3
---

# Connect to SurrealDB via Ngrok tunnel

# Connect to SurrealDB via Ngrok tunnel

This guide will walk you through connecting a local SurrealDB instance to the internet using Ngrok, making it accessible remotely. 
[Ngrok](https://ngrok.com/) is a cross-platform application that allows developers to expose their local web servers to the internet. It hosts a local web server on its own sub-domain and makes your local development box available on the internet through Tunneling.

This setup is beneficial for remote development and quick testing of applications built locally on SurrealDB. 

## Prerequisites

This guide assumes the following:

> - You have [SurrealDB installed](https://surrealdb.com/install) and a [fresh instance of SurrealDB running.](/docs/1.1.x/introduction/start)

Reviewer: ```suggestion
- You have [SurrealDB installed](/docs/1.1.x/installation/overview) and a [fresh instance of SurrealDB running.](/docs/1.1.x/introduction/start)
```

## Connect to SurrealDB via Ngrok tunnel

This comprehensive guide will walk you through connecting a local SurrealDB instance to the internet using Ngrok, making it accessible remotely. 
Ngrok is **a cross-platform application that allows developers to expose their local web servers to the internet**. It hosts a local web server on its own sub-domain and makes your local development box available on the internet through Tunneling.

This setup is beneficial for remote development and quick testing of applications built locally on SurrealDB. 

### Prerequisites

This guide assumes the following:

> - You have [SurrealDB installed](https://surrealdb.com/install) and a [fresh instance of SurrealDB running.](https://docs.surrealdb.com/docs/introduction/start)

Reviewer: Within docs, we use the relative paths  so `https://docs.surrealdb.com/docs/introduction/start` will be `/docs/introduction/start`

- Downloaded [Ngrok](https://ngrok.com/download) to be used as a tunnel.

> Tunneling refers to the process of using a network protocol to encapsulate a different payload protocol, enabling data to pass securely through a network or the internet
> 

### Steps

1. **Start SurrealDB Instance**
    

>     Open your command line or terminal and run the following command to [start SurrealDB](https://docs.surrealdb.com/docs/introduction/start/).

Reviewer: Same here as well

>     ```bash
>     
>     surreal start memory -A --auth --user root --pass root
>     ```

Reviewer: ```suggestion
    ```bash
    surreal start memory -A --auth --user root --pass root
    ```
```

    
    We use the default username and password “root”.
    
2. **Set Up Ngrok Tunnel**
    
    Open another command line or terminal window (do not close the SurrealDB one) and run the following command to expose SurrealDB’s default port (8000) to the internet:
    

>     ```bash
>     
>     ngrok tcp 8000
>     
>     ```

Reviewer: ```suggestion
    ```bash
    ngrok tcp 8000
    ```
```

    
    Note the forwarding address provided by Ngrok (e.g., **`0.tcp.in.ngrok.io:14033`**).
    
    Keep this address handy as we will use it in the next step.
    
3. **Connect to SurrealDB via the Ngrok Address**
    
    Now, let's [connect to your SurrealDB instance](https://docs.surrealdb.com/docs/cli/sql/) using the forwarding address provided by Ngrok.
    
    Run the following command in a new terminal window/tab, replacing **`[ngrok-address]`** with the actual Ngrok forwarding address you noted earlier:
    

>     ```bash
>     
>     surreal sql --conn ws://[ngrok-address] --user root --pass root --ns test --db test --pretty
>     
>     ```

Reviewer: ```suggestion
    ```bash
    surreal sql --conn ws://[ngrok-address] --user root --pass root --ns test --db test --pretty
    ```
```

> 4. **Verify the Connection**
>     
>     Finally, let’s ensure the connection is properly set up by running a test query.
>     With the help of the `CREATE` statement, [create a new record](https://docs.surrealdb.com/docs/surrealql/statements/create). 

Reviewer: ```suggestion
4. **Verify the Connection**
    
    Finally, let’s ensure the connection is properly set up by running a test query.
    With the help of the `CREATE` statement, [create a new record](/docs/surrealql/statements/create). 
```
